In [1]:
import pandas as pd
import numpy as np
import os
import glob
import matplotlib as mpl
import matplotlib.pyplot as plt
from matplotlib import cm
%matplotlib inline
import math

In [2]:
#Electronic Energy / H = [ Reactant, TS1, Intermediate, TS2, Product ]
raw_e = [-980.705872, -980.68265, -980.684249, -980.681684, -980.685145]

In [3]:
# Electronic Energy Relative to Reactant / H 
e_hart = np.array(raw_e) - np.array(raw_e[0])
e_hart

array([ 0.      ,  0.023222,  0.021623,  0.024188,  0.020727])

In [4]:
# Conversion Factors
hart2kcal = 627.509
kcal2j = 4.184

In [5]:
# Electronic Energy Relative to Reactant / kcal/mol
e_kcal = e_hart * hart2kcal
e_kcal

array([  0.        ,  14.572014  ,  13.56862711,  15.17818769,  13.00637904])

In [6]:
# Electronic Energy Relative to Reactant / j/mol
e_j = e_kcal*kcal2j
e_j

array([  0.        ,  60.96930657,  56.77113582,  63.5055373 ,  54.41868992])

In [7]:
# Constants
h = 6.62607004*10**(-34) # Js
kb = 1.3807*10**(-23) #J/K
T = 300 # K 
R= 8.3144598 #J  K−1 mol−1
kbT_h = (kb*T)/h # s-1

Transition State Theory

In [8]:
%%latex
$k(T) = \kappa(T)\frac{k_bT}{h} \mathrm{exp} \biggl(-\frac{\Delta G^\ddagger}{RT}\biggr)$

<IPython.core.display.Latex object>

Tunnelling correction

In [9]:
%%latex
$\kappa(T) = 1 + \frac{1}{24} {\biggl(\frac{h \mathrm{Im}(v^\ddagger)}{k_bT}\biggr)}^2$

<IPython.core.display.Latex object>

Values:

TS1 Electronic Energy (Relative to reactant), $\Delta$E$_1$$^\ddagger$ = 14.57 kcal/mol or 60.97 j/mol

TS1 Imaginary Frequency, 966.627 cm$^{-1}$ or 96662.7 m$^{-1}$

Intermediate Electronic energy (Relative to reactant), $\Delta$E$_{Int}$ = 13.57 kcal/mol or 56.77 j/mol

TS2 Electronic Energy (Relative to reactant), $\Delta$E$_2$$^\ddagger$ = 15.18 kcal/mol or 63.51 j/mol

TS2 Imaginary Frequency, 863.314 cm$^{-1}$ or 86331.1 m$^{-1}$

In [33]:
# Tunneling for TS1
tun1 = 1 + (1/24) * ((h*96662.7)/kb*T)**2 
print(tun1, 's^2 m^-2')

1.0000000806979508 s^2 m^-2


In [31]:
# Forward rate without tunnelling correction
k_nocorr1 = kbT_h*math.exp(-( 60.96930657/(R*T)) )
print('%.2E' % k_nocorr1 , 's^-1')

6.10E+12 s^-1


In [34]:
# Forward rate with tunnelling correction
k_corr1 = tun1*kbT_h*math.exp(-( 60.96930657/(R*T)) )
print('%.2E' % k_corr1 , 's m^-2')

6.10E+12 s m^-2


In [35]:
# Tunnelling for TS2
tun2 = 1 + (1/24) * ((h* 86331.1)/kb*T)**2 
print(tun2, 's^2 m^-2')

1.000000064369363 s^2 m^-2


In [36]:
# Forward rate without tunnelling correction
k_nocorr2 = kbT_h*math.exp(-( (63.51-56.77)/(R*T)) )
print('%.2E' % k_nocorr2 , 's^-1')

6.23E+12 s m^-2


In [37]:
k_corr2 = tun2 * kbT_h*math.exp(-( (63.51-56.77)/(R*T)) )
('%.2E' % k_corr2, 's m^-2')

('6.23E+12', 's m^-2')